lib/fluent/plugin/out_mongo.rb

```diff
@@ -213,26 +219,33 @@ def collect_records(chunk)
         if date_keys
           date_keys.each { |date_key|
             begin
-              date_value = record[date_key]
+              date_key_accessor = @accessors[date_key.to_s]
+              date_value = date_key_accessor.call(record)
+              if date_value.to_i.to_s == date_value
+                date_value = date_value.to_i
+              elsif date_value.to_f.to_s == date_value
+                date_value = date_value.to_f
+              end
               case date_value
               when Fluent::EventTime
-                record[date_key] = date_value.to_time
+                value_to_set = date_value.to_time
               when Integer
-                record[date_key] = if date_value > 9999999999
+                value_to_set = if date_value > 9999999999
                                      # epoch with milliseconds: e.g. javascript
                                      Time.at(date_value / 1000.0)
                                    else
                                      # epoch with seconds: e.g. ruby
                                      Time.at(date_value)
                                    end
               when Float
-                record[date_key] = Time.at(date_value)
+                value_to_set = Time.at(date_value)
               else
-                record[date_key] = Time.parse(date_value)
+                value_to_set = Time.parse(date_value)
               end
+              date_key_accessor.set(record, value_to_set)
             rescue ArgumentError
               log.warn "Failed to parse '#{date_key}' field. Expected date types are Integer/Float/String/EventTime: #{record[date_key]}"
```

Comment:
Use `date_value` instead of `record[date_key]`

Reply:
Sorry, I don't understand your comment. I use `value_to_set` because contains the new value. What do you mean with 'Use date_value'?

Reply:
`date_value` is:
```ruby
              date_value = date_key_accessor.call(record)
```

ref: https://github.com/fluent/fluent-plugin-mongo/pull/148/files#diff-1ed273a50004a1a0d6d61cf6c3e367e641c3962dff65df7aa836eb8f12fb7c84R227

Reply:
With the refactor of the latest commit (ref: [https://github.com/fluent/fluent-plugin-mongo/pull/148#discussion_r526775541](https://github.com/fluent/fluent-plugin-mongo/pull/148#discussion_r526775541)) I think this was implemented.
Let me know if i need to implement more changes

Reply:
I mean that can we use `date_value` in string interpolation?

```ruby
            rescue ArgumentError
              log.warn "Failed to parse '#{date_key}' field. Expected date types are Integer/Float/String/EventTime: #{date_value}"
```